thoth-api/src/model/location/crud.rs

```diff
@@ -0,0 +1,182 @@
+use super::{
+    Location, LocationField, LocationHistory, LocationPlatform, NewLocation, NewLocationHistory,
+    PatchLocation,
+};
+use crate::graphql::model::LocationOrderBy;
+use crate::graphql::utils::Direction;
+use crate::model::{Crud, DbInsert, HistoryEntry};
+use crate::schema::{location, location_history};
+use crate::{crud_methods, db_insert};
+use diesel::{ExpressionMethods, QueryDsl, RunQueryDsl};
+use thoth_errors::{ThothError, ThothResult};
+use uuid::Uuid;
+
+impl Crud for Location {
+    type NewEntity = NewLocation;
+    type PatchEntity = PatchLocation;
+    type OrderByEntity = LocationOrderBy;
+    type FilterParameter1 = LocationPlatform;
+    type FilterParameter2 = ();
+
+    fn pk(&self) -> Uuid {
+        self.location_id
+    }
+
+    fn all(
+        db: &crate::db::PgPool,
+        limit: i32,
+        offset: i32,
+        _: Option<String>,
+        order: Self::OrderByEntity,
+        publishers: Vec<Uuid>,
+        parent_id_1: Option<Uuid>,
+        _: Option<Uuid>,
+        location_platform: Option<Self::FilterParameter1>,
+        _: Option<Self::FilterParameter2>,
+    ) -> ThothResult<Vec<Location>> {
+        use crate::schema::location::dsl;
```

Comment:
Perhaps this could have been `use crate::schema::location::dsl::*` as I don't think there are name clashes (?)